src/libraries/System.Private.CoreLib/src/System/Number.Parsing.cs

```diff
@@ -1725,6 +1757,8 @@ internal static unsafe ParsingStatus TryParseDecimal(ReadOnlySpan<char> value, N
             return ParsingStatus.OK;
         }
 
+        internal static bool SpanStartsWith(ReadOnlySpan<char> span, char c) => span.Length > 0 && span[0] == c;
```

Comment:
FWIW, the JIT today will generate better code if you instead write this as:
```suggestion
        internal static bool SpanStartsWith(ReadOnlySpan<char> span, char c) => !span.IsEmpty && span[0] == c;
```
https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIGYACMhgYQYG8aHuGBLAOwwwo/bABsm5JA2AQIEgMoAHbPwUZsUDLgDqvDAAtyACgBKMbABMA8vzEBPZaoA8YA5oB8DXCv5oGbpoBAJQMALxePqoAdAAyMPwA5oYMXgAMDABkmd6+ANppALrhYQEA3Fw8AkIi4pLSsvIMTmoaWrr6BqRmFjZ2jr6u7lCRvv6BUCHhXgCEUfzRAJK4AKL4Shj2WTnzBcVhpWAV1AC+QA===